# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value. Try using both the bootstrapping and the frequentist statistical approaches.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
+ Formulas for the Bernoulli distribution: https://en.wikipedia.org/wiki/Bernoulli_distribution

In [10]:
import pandas as pd
import numpy as np
from scipy import stats
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [5]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [6]:
# number of callbacks for black-sounding names
sum(data[data.race=='w'].call)

235.0

In [7]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

<div class="span5 alert alert-success">
<p>Your answers to Q1 and Q2 here</p>
</div>

## Question 1

I would use a z-test, and CLT does apply.

## Question 2

My null hypothesis would be that whites and blacks have similar response rates.

My alternate hypothesis would be that whites and blacks do not have similar response rates.

In [8]:
w = data[data.race=='w']
b = data[data.race=='b']

## Question 3

Compute margin of error, confidence interval, and p-value. Try using both the bootstrapping and the frequentist statistical approaches.


In [16]:
w_callback = np.mean(w['call'])
b_callback = np.mean(b['call'])
callback_diff = w_callback - b_callback
callback_mean = np.mean(data.call)

call_diff = np.empty(len(w))
for i in range(len(w)):
    call_diff[i] = w.call.iloc[i] - b.call.iloc[i]

w_call_var = np.var(w['call'])
b_call_var = np.var(b['call'])
call_var = np.var(data.call)
call_vars = (w_call_var + b_call_var)
call_varss = (callback_mean*(1-callback_mean))

margin_of_error = (1.96 * call_var / len(data)) ** .5

p = (np.sum(call_diff > 0) + np.sum(call_diff < 0)) / (len(data)/2)

print("Difference between white and black applicants:", callback_diff)
print('Mean call rate for all applicants:',round(callback_mean,4))
print('Mean call rate for white applicants:',round(w_callback,4))
print('Mean call rate for black applicants:',round(b_callback,4))
print('Margin of error:', round(margin_of_error,4))
print('95% CI for all applicants:',round(callback_mean-margin_of_error,4),',',round(callback_mean + margin_of_error,4))
print('p-value:', p)
print('Z-score:',(0-callback_diff)/(call_var/len(data))**.5)
#Cannot tell distribution, I suspect it is bimodal


Difference between white and black applicants: 0.03203285485506058
Mean call rate for all applicants: 0.0805
Mean call rate for white applicants: 0.0965
Mean call rate for black applicants: 0.0645
Margin of error: 0.0055
95% CI for all applicants: 0.075 , 0.086
p-value: 0.09774127310061602
Z-score: -8.216918916654633


<div class="span5 alert alert-success">
<p> Your answers to Q4 and Q5 here </p>
</div>

## Conclusion

We do see that “white” applicants are asked to come in for interviews more often than “black” applicants. The expected call rate for all applicants is: 0.0805 While the expected call rate for white applicants is: 0.0965 While the expected call rate for black applicants is: 0.0645

Meanwhile the average black applicant is beyond the lower bound of our 95% confidence interval. The margin of error is 0.0055. The 95% CI for all applicants is 0.075, 0.086.

Hypothesis testing was also done. The null hypothesis was that racism did not exist where our difference in callback rate was zero on average, and our alternative hypothesis was that racism did exist and difference in callback is not zero. We got a p-value of 0.097741. This is above the standard p value that denotes statistical significance, thus we have to say that some racial discrimination exists. Average education level and work experience was same for both applicants. Also, females are asked to come in for interviews in greater percentages.